plugins/polio/api/vaccines/stock_management.py

```diff
@@ -78,11 +75,169 @@ class VaccineStockManagementReadWritePerm(GenericReadWritePerm):
     write_perm = permission.POLIO_VACCINE_STOCK_MANAGEMENT_WRITE
 
 
+class StockManagementCustomFilter(filters.BaseFilterBackend):
+    def filter_queryset(self, request, queryset, _view):
+        country_id = request.GET.get("country_id")
+        vaccine_type = request.GET.get("vaccine_type")
+
+        if country_id:
+            queryset = queryset.filter(country_id=country_id)
+        if vaccine_type:
+            queryset = queryset.filter(vaccine=vaccine_type)
+
+        current_order = request.GET.get("order")
+
+        if current_order:
+            if current_order == "country_name":
+                queryset = queryset.order_by("country__name")
+            elif current_order == "-country_name":
+                queryset = queryset.order_by("-country__name")
+            elif current_order == "vaccine_type":
+                queryset = queryset.order_by("vaccine")
+            elif current_order == "-vaccine_type":
+                queryset = queryset.order_by("-vaccine")
```

Comment:
`rest_framework.filters.OrderingFilter` gives you [ordering for free](https://www.django-rest-framework.org/api-guide/filtering/#orderingfilter).

Reply:
@kemar I know but the names do not match and I couldn't figure out how to do name mapping or whatever it's called. If you have an example it would be welcome

Reply:
@mathvdh ok then, I've got no better solution right now.